pkg/image/mapping.go

```diff
@@ -25,7 +26,20 @@ func ParseTypedImage(image string, typ v1alpha2.ImageType) (TypedImage, error) {
 	if err != nil {
 		return TypedImage{}, err
 	}
-	return TypedImage{ref, typ}, nil
+	t := TypedImage{ref, typ}
+	return t.SetDefaults(), nil
+}
+
+// SetDefaults sets the default values for TypedImage fields
+func (t TypedImage) SetDefaults() TypedImage {
+	if len(t.Ref.Tag) == 0 {
+		if len(t.Ref.ID) > 13 {
```

Comment:
Magic numbers and random range below as well. A comment would be nice.

Reply:
Thanks! I will add a comment. The number selection was arbitrary, get getting a tag from the first 6 characters from the digest. Only check that it's great than 13 to avoid an out-of-range panic.

Reply:
Decided to remove this altogether and just made a helper function to validate and get the partial digest so the solution does not diverge in these two places.